staging/kos/cmd/attachment-tput-driver/main.go

```diff
@@ -1297,3 +1309,33 @@ func withRetries(thunk func() bool, minWait, maxWait time.Duration, stopCh <-cha
 	}
 	return true
 }
+
+func printCSVOpsTimes(opsTimes map[uint64]threadOpsSchedule, rate float64, distribution, outputDir string) (err error) {
+	opsTimesFilename := filepath.Join(outputDir, "attachments-ops-times.csv")
+	var opsTimesCSVFile *os.File
+	opsTimesCSVFile, err = os.Create(opsTimesFilename)
+	if err != nil {
+		return
+	}
+	defer func() {
+		cerr := opsTimesCSVFile.Close()
+		if err == nil {
+			err = cerr
+		}
+	}()
+	header := fmt.Sprintf("\"distribution: %s, rate: %g\"\n\"cell(i, j > 0) = dt in seconds at which ops j for thread i must take place wrt start of operations\"\n", distribution, rate)
+	if _, err = opsTimesCSVFile.Write([]byte(header)); err != nil {
+		return
+	}
+	for thd := 0; thd < len(opsTimes); thd++ {
```

Comment:
This loop only works if the map is equivalent to a slice --- the valid indices are 0 (inclusive) through len (exclusive).  Why use a map instead of a slice?

Reply:
49677ba changes the data structure that stores ops times to a single, flat slice where ops i of thread j is at index `(i*numThreads + j)`.